### ISSUE TITLE
Embedded player broken in contexts that do not allow localStorage

### ISSUE THREAD

**Issue by scanlime** [Chocobozzz/PeerTube#4412]
**Describe the current behavior**

On some web sites and some browsers, the update to PeerTube 3.4.0 broke video embeds. The videos display as a black rectangle with the text "Sorry" "The player is not compatible with your web browser. Please try latest Firefox version".

This message appears on current versions of both Chrome and Firefox.

**Steps to reproduce:**

The underlying error I'm seeing is that a "TypeError: p.getItem is not a function" caused by localStorage being undefined. It seems like this can happen for multiple reasons, including:

1. Opening the outer page in a private browsing window
2. Blocking third-party cookies in the browser

**Describe the expected behavior**

Expect embeds to keep working on browsers and pages where they previously worked.

**Additional information**

* PeerTube instance: 
  * URL: https://diode.zone  (sample broken embed: https://scanlime.org/2019/02/scanlime047-icestudio-led-matrix-driver-part-3/)
  * version: 3.4.0
  * NodeJS version: v12.22.6
  * Ffmpeg version: 4.2.4-1ubuntu0.1

* Browser name, version and platforms on which you could reproduce the bug: Firefox on Linux, Chrome on ChromeOS
* Link to browser console log if relevant:
```
video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:17371 Cannot init embed. TypeError: p.getItem is not a function
    at Function.load (video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:16638)
    at Ae.<anonymous> (video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:16989)
    at Generator.next (<anonymous>)
    at video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:3597
    at new Promise (<anonymous>)
    at r (video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:3574)
    at Ae.init (video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:16988)
    at Function.<anonymous> (video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:16848)
    at Generator.next (<anonymous>)
    at video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:3597
(anonymous) @ video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:17371
Promise.catch (async)
(anonymous) @ video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:17369
(anonymous) @ video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:17375
(anonymous) @ video-embed.cb94f3ad1d26ba89f80c.bundle.js:formatted:17377
```

* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

**Comment by merlin1608** [comment]
Hi , same behaviour on my website, all the videos are inaccessible with this message " "Sorry" "The player is not compatible with your web browser. Please try latest Firefox version" tested with firefox, chrome on Linux and Windows
I don't know if there is a turnaround or if a have to wait for a patch on peertube.

**Comment by pietsch** [comment]
Same problem here. Some observations:

- Firefox 92 on Linux: not affected.
- Stock browser on LineageOS: not affected.
- Privacy browser for Android: not affected after you enable JavaScript and DOM storage for this site. 
- Users with Firefox 92 were affected if they viewed our Drupal site before PeerTube were upgraded from 3.3.0 to 3.4.0. Workaround: Viewing our Drupal site with a fresh Firefox profile.

**Comment by ulfi** [comment]
Same issue on my side:
- Firefox 92.0 on Linux (openSUSE Tumbleweed)
- Works also on Falkon 3.1.0 at desktop and Fenec 92.1.1 at Android
- Works on Firefox 91.x before update
- Effected pages iframe: https://lug-vs.org/
- My peertube account is an empty page: https://tube.sp-codes.de

**Comment by Oymate** [comment]
ulfi's 2nd link is working. But joinpeertube embedded video isn't

**Comment by jata-PT** [comment]
The same story here. Many very angry friends, using embedding video(s). Suddenly after the PeerTube upgrade (3.3.0->3.4.0) looking just on the black box with the 'Sorry...' message. We had to revert all system back with using of the old snapshot.

**Comment by aschrijver** [comment]
Since not so long I get the message everywhere, but with a different JS console error: 

```
Cannot init embed. DOMException: The operation is insecure. video-embed.0c4c878d45b60469b924.bundle.js:2:343537
    <anonymous> https://share.tube/client/standalone/videos/video-embed.0c4c878d45b60469b924.bundle.js:2
    (Async: promise callback)
    <anonymous> https://share.tube/client/standalone/videos/video-embed.0c4c878d45b60469b924.bundle.js:2
    <anonymous> https://share.tube/client/standalone/videos/video-embed.0c4c878d45b60469b924.bundle.js:2
    <anonymous> https://share.tube/client/standalone/videos/video-embed.0c4c878d45b60469b924.bundle.js:2
```

This is embedded from [spritelyproject.org](https://spritelyproject.org) landing page. Also the ActivityPub conference videos on [redaktor.me](https://redaktor.me/apconf/) are all broken.

Using Firefox 98.0.2 on Ubuntu 18.04

@Chocobozzz does this need a new issue, or does this look like something on the browser side that's fishy?

**Comment by Chocobozzz** [comment]
> @Chocobozzz does this need a new issue, or does this look like something on the browser side that's fishy?

Hi,

See https://github.com/dexie/Dexie.js/pull/1465. Transcoding videos to HLS should prevent this issue from happening

**Comment by abcdw** [comment]
Hi everyone! I have a similiar problem on icecat 102.6 with peertube 5.0.1. Have a stream from diode.zone embeded with iframe on https://trop.in/stream. 

![image](https://user-images.githubusercontent.com/1218615/213922291-38ecb33f-1b98-4ee3-a4d2-1d41963ca768.png)

Is it the same issue?

In console I found a message, which can be related:
```
Request to access cookie or storage on “https://diode.zone/videos/embed/e7098d6e-0314-4ff2-835f-d01bab54dad5” was blocked because we are blocking all third-party storage access requests and content blocking is enabled.
```

Please, let me know if I need to provide any additional information.

**Comment by aschrijver** [comment]
> Is it the same issue?

Yes. It is showing that in the latest Firefox browsers. When opening such videos directly on the peertube instance they work fine.

**Comment by Chocobozzz** [comment]
@aschrijver The bug for webtorrent videos is not fixed and won't be as you can see in https://github.com/dexie/Dexie.js/pull/1465#issuecomment-1380130675
But we plan to remove webtorrent support in the future: https://github.com/Chocobozzz/PeerTube/issues/5465

**Comment by abcdw** [comment]
@Chocobozzz got it, thank you for the links!
